src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -0,0 +1,1567 @@
+    }
+
+    // Sort these differences and find the index 1/10 in from the left
+    // constrained betweeen indexes 1 and 3 inclusive; 
+    // return the absolute value of this pair's difference
+    Arrays.sort(tmpData);
+    int tmpIdx = (int) (numVals * 0.1);
+    if (tmpIdx < 1) {
+      tmpIdx = 1;
+    }
+    if (tmpIdx > 3) {
+      tmpIdx = 3;
+    }
+    return Math.abs(tmpData[tmpIdx]);
+  }
+
+  // Almost certainly, these are pre-existing functions somewhere???
+  // No error checking done here ...
+  
+  /**
+   * Get a mean value over a slice of an array
+   *
+   * @param arr - Array to calculate mean
+   * @param start - start index of array
+   * @param end - end index of array (not included!)
+   * @return - mean value of slice
+   */
+  public static final double getSliceMean(double[] arr, int start, int end) {
+    double sum = 0.0;
+    for (int i = start; i < end; i++) {
+      sum += arr[i];
+    }
+    return ((double) sum / (end - start));
+  }
+
+  /**
+   * Get a standard deviation value over a slice of an array
+   *
+   * @param arr - Array to calculate standard deviation
+   * @param start - start index of array
+   * @param end - end index of array (not included!)
+   * @return - standard deviation value of slice
+   */
+  public static final double getSliceStd(double[] arr, int start, int end) {
+    double var = 0.0;
+    double mean = getSliceMean(arr, start, end);
+    for (int i = start; i < end; i++) {
+      var += (arr[i] - mean) * (arr[i] - mean);
+    }
+    return Math.sqrt((double) var / (end - start));
+  }
+
+  /**
+   * Least-squares linear fit of Y on X
+   *
+   * @param y - array of dependent values
+   * @param x - array of independent values
+   * @param numObs - length of array
+   * @return - slope, intercept, r2 and t value of least-squares linear regr.
+   */
+  public static final double[] linearFit(double[] y, double[] x, int numObs) {
+    // TODO: Probably should be replaced with generic linear fitting algorithm
+    // but kept in here for comparison purposes
+
+    double sxx = 0.0;
+    double sxy = 0.0;
+    double syy = 0.0;
+
+    double meanX = getSliceMean(x, 0, numObs);
+    double meanY = getSliceMean(y, 0, numObs);
+    for (int i = 0; i < numObs; i++) {
+      sxx += (x[i] - meanX) * (x[i] - meanX);
+      syy += (y[i] - meanY) * (y[i] - meanY);
+      sxy += (y[i] - meanY) * (x[i] - meanX);
+    }
+
+    if (sxx < 0.00001) {
+      // Handle exception
+      // String msg = "Error: divided by 0 in linear fit\n";
+      // throw new Exception(msg);
+    }
+
+    double slope = sxy / sxx;
+    double intercept = meanY - meanX * slope;
+    double r2;
+    if (syy < 0.00001) {
+      r2 = 0.0;
+    } else {
+      r2 = (sxy * sxy) / (sxx * syy);
+    }
+    double denom = r2 == 1.0 ? 0.00001 : 1.0 - r2;
+
+    // TODO: What is t?
+    double t = Math.sqrt(r2 * (numObs - 2) / denom);
+
+    return new double[]{slope, intercept, r2, t};
```

Comment:
Because of this line, I think you want to make linearFit an inner class that holds on to these, and just have 1 as a member variable that you can call over and over.

```
class linearFit() {
    double coeff[] = new double[4];
    double[] fit(double[] y, double[] x, int numObs) {
          ...
          return coeff;
    }
}

coeff = this.linearFit.fit(x, y, numObs);
```

Although returning the internal array is mildly cheesy.  Real java guys would make separate accessors for each element.